molo/core/models.py

```diff
@@ -676,21 +647,23 @@ def save(self, *args, **kwargs):
         languages = Languages.for_site(self.get_site()).languages
         if (languages.filter(
                 is_main_language=True).exists() and
-                not self.languages.exists()):
+                not self.language):
             language = languages.filter(
                 is_main_language=True).first()
             LanguageRelation.objects.create(
                 page=self,
                 language=language)
-
+            self.language = language
+            self.save()
         return response
 
     def move(self, target, pos=None):
         super(TranslatablePageMixinNotRoutable, self).move(target, pos)
 
-        if hasattr(self, 'translations'):
-            for p in self.translations.all():
-                p.translated_page.move(target, pos='last-child')
+        if hasattr(self, 'translated_pages'):
+            for p in self.translated_pages.all():
+                if not Page.objects.filter(pk=self.pk).first():
```

Comment:
For example, the english article is a translated page for spanish and french article, when we move each of them over, it's going to try and move all of the translated articles over as well, which means it going to try and move the english article twice, so we check here to only move it if it doesn't already exist

Reply:
I'm afraid I'm still confused because we're looking for any objects with the same pk as the one we're currently working with. So wouldn't this always be true?

Reply:
The move method gets called for each translated page. So when the page's move method is called, and the translated page being moved is self, it checks whether self exists in the new destination

Reply:
Oh wait, I see what you're saying. Let me go over it again